### PR TITLE
Fix (minor) deprecations

### PR DESCRIPTION
**Pull request by tlambert03**
# Description
this just fixes a couple deprecation warnings that show up in our tests.  perhaps most importantly `from collections import Iterable` will stop working in python3.8

@sofroniewn, the comparison in this line is deprecated ... but it wasn't immediately clear to me what the goal of the logic is:
https://github.com/napari/napari/blob/4096e71b7e1fa041a62f4ac2f6853fba60c93e52/napari/layers/base/base.py#L613
 
```
napari/layers/image/tests/test_image.py::test_negative_rgba_image
  /Users/talley/Dropbox (HMS)/Python/forks/napari/napari/layers/base/base.py:614
  DeprecationWarning: elementwise comparison failed; this will raise an error in the future.
    if value is not None and not np.all(value == (None, None)):
```

is it just making sure that, if `value` is an array, at least one value is not null?

### CODE REVIEW

File: napari/layers/base/base.py [hunk 1]
```diff
@@ -610,6 +610,7 @@ def get_message(self):
 
         value = self._value
 
+        # FIXME: deprecation warning... but logic unclear
```
Comment (sofroniewn) [inline napari/layers/base/base.py]: yeah, this isn't so nice. Some layers return value as a singleton, hence the `is not None`, but some layers like `Shapes` can return a tuple, I think the following would work

```python
if value is not None and not np.all([v == None for v in value])
```

**Review by sofroniewn — COMMENTED** [review]
Thanks for doing this @tlambert03. I left a suggestion on that strange line. Maybe we can fix this before merge. I'm surprised there aren't more warnings, or does this only fix a subset. Would be good to have them all fixed at some point

File: napari/layers/base/base.py (hunk 1 above)
Comment (tlambert03) [reply napari/layers/base/base.py]: actually, looks like `value` can be: `None`, a scalar, a `tuple`, and even a`tuple` with a nested numpy `array` (for rgb_pyramid).   `for v in value` will throw an error if `value` is not iterable, so that won't work.  Do we just want to check whether it's a non-null scalar, or an iterable with no null values?  (or an iterable with at least one non-null value?)

Comment (sofroniewn) [reply napari/layers/base/base.py]: that sounds reasonable for now, then we might want to refactor the layers so the output of value is a little more constrained

**Review by jni — APPROVED** [review]

**Comment by jni** [comment]
I'm happy to merge as is and then constrain the possible return types of `value`. Even in the suggested fix, the logic is hard to follow... Additionally, that whole machinery might change when we add support for world coordinates...

**Comment by tlambert03** [comment]
I tweaked the logic of `get_message` very slightly.  I verified that the behavior is still the same with RGB, grayscale, and RGB images.  Hopefully it's a little bit clearer, and the deprecation warning is gone.  If you are ok with it, feel free to merge.

**Comment by tlambert03** [comment]
not sure what's [going on with black](https://github.com/napari/napari/pull/800/checks?check_run_id=351350474)... closing and reopening to force rebuild

**Comment by kne42** [comment]
@tlambert03 are you able to force a rebuild by clicking on "details" next to the check?

**Comment by tlambert03** [comment]
yeah, but neither that method nor the other methods are fixing it.  almost seems like a problem with the python install on cirrus?

**Comment by kne42** [comment]
yeah, that's bizarre. the versions of the installed packages haven't changed either

**Comment by tlambert03** [comment]
looks like it finally succeeded at around 6:05 EST...  🤷‍♂️merging!

**Comment by sofroniewn** [comment]
Nice, I triggered a manual re-run at that time, and I guess it had been fixed by our CI :-)
